lib/KDropdownMenu.vue

```diff
@@ -1,40 +1,28 @@
 <template>
-
-  <div class="dib">
-    <!-- Added because the trigger could not be a vue element smh-->
-    <div
-      ref="buttonContainer"
-      class="button-container dib"
-    >
-      <slot name="button"></slot>
-      <KButton
-        v-if="!$slots.button"
```

Comment:
The change here fundamentally changes the API of the component by removing the allowance of a named `buttons` slot.

After looking into Kolibri, I see that we don't use that slot anywhere ( or in Studio - where we dont' use this component at all )

I don't think we actually need this part of the API - but it's worth noting in any case (and in our CHANGELOG) that the API for this component has changed.

Reply:
I think that it's actually good to remove this from the API in this case because the component (I believe) is already designed with good consideration for our design system guidance on buttons and offering the slot makes it easier to use the component while incorrectly styling the button.

Reply:
Relevant: https://github.com/learningequality/kolibri-design-system/issues/164

Long story short, I think `KDropdownMenu` should not exist in it's current form. However, deprecating it could only happen in a "major" (technically minor in the [semver](https://semver.org/) sense) bump because this would truly be a breaking change.

In this case I agree with @nucleogenesis that we can drop the API because we know with high confidence that we won't break apps for library users. Once KDS gets more widely used (and probably once we call it 1.0.0) we'll want to get a less cavalier about this kind of change